app/scripts/app.js

```diff
@@ -56,12 +61,17 @@ angular.module('yourStlCourts').config(function($stateProvider, $urlRouterProvid
         court: {value: undefined}
       },
       resolve: {
-        court: function ($stateParams,Courts) {
-          if ($stateParams.courtId != ""){
-            return Courts.findById($stateParams.courtId)
-          }else{
-            return null;
-          };
+        court: function ($stateParams,Courts,Error) {
+          if ($stateParams.courtId == "") {
+            throw Error.errorObject(ErrorCode.BAD_REQUEST, "No Court was found with the url you provided.");
+          } else {
+            return (Courts.findById($stateParams.courtId)
+              .then(function (court) {
+                return court;
+              },function(){
```

Comment:
Also, with promises, try to keep ".then" and ".catch" on the same line as the function call that returns the promise. That is typically how it's laid out in JS-land.